index.ts

```diff
@@ -303,9 +310,24 @@ export class Project {
       if (stat.isDirectory()) {
         this.hardLinkContents(path.join(target, name), path.join(destination, name));
       } else {
-        fs.ensureLinkSync(path.join(target, name), path.join(destination, name));
+        this.hardLinkFile(path.join(target, name), path.join(destination, name));
+      }
+    }
+  }
+
+  private hardLinkFile(source: string, destination: string) {
+    if (this.usingHardLinks) {
+      try {
+        fs.ensureLinkSync(source, destination);
+        return;
+      } catch (err: any) {
+        if (err.code !== 'EXDEV') {
+          throw err;
+        }
+        this.usingHardLinks = false;
       }
     }
+    fs.copySync(source, destination);
```

Comment:
thoughts on using `fs.copyFileSync(source, destination, fs.constants.COPYFILE_FICLONE)` ?

* available since Node 10.
* on filesystems that support it, it will be a copy-on-write clone (APFS – basically all our apple users, BTRS, XFS)

Reply:
Actually `fs.constants.COPYFILE_EXCL | fs.constants.COPYFILE_FICLONE` may be the most appropriate flag

Reply:
Done